cmd/checker/checker.go

```diff
@@ -0,0 +1,59 @@
+package checker
+
+import (
+	"kool-dev/kool/cmd/builder"
+	"log"
+)
+
+// Checker defines the check kool dependencies method
+type Checker interface {
+	CheckKoolDependencies() (string, error)
```

Comment:
This interface should only return an `error`. One more comment about this below on its implementation for DefaultChecker.

Reply:
Maybe now that this moved onto its own package the name of this function could become something like `VerifyDependencies`